src/Container/ApplicationFactory.php

```diff
@@ -77,19 +63,13 @@ public function __invoke(ContainerInterface $container)
             ? $container->get(RouterInterface::class)
             : new FastRouteRouter();
 
-        $finalHandler = ! empty($config['zend-expressive']['raise_throwables'])
-            ? $this->marshalNoopFinalHandler($container)
-            : $this->marshalLegacyFinalHandler($container, $config);
+        $delegate = $this->marshalNoopFinalHandler($container);
```

Comment:
The legacy final handler implementation can be retired with a major version bump. This will be a BC break in behavior, but can be fixed by pointing the `NoopFinalHandler` service to a factory for creating a `FinalHandler` instance if desired.